plugins/router/template/router.go

```diff
@@ -252,6 +280,15 @@ func (r *templateRouter) AddEndpoints(id string, endpoints []Endpoint) {
 	r.state[id] = frontend
 }
 
+// cleanUpServiceAliasConfig performs any necessary steps to clean up a service alias config before deleting it from
+// the router.  Right now the only clean up step is to remove any of the certificates on disk.
+func (r *templateRouter) cleanUpServiceAliasConfig(cfg *ServiceAliasConfig) {
+	err := r.certManager.deleteCertificatesForConfig(cfg)
+	if err != nil {
+		glog.Errorf("Error deleting certificates for route %s, the route will still be deleted but files may remain in the container: %v", cfg.Host, err)
```

Comment:
Ok.  That gives me a good way to eliminate the todo about writing certs every time we write config: `//TODO: better way so this doesn't need to create lots of files every time state is written, probably too expensive`. No need to rewrite an existing, matching file.  

Re: overwriting, the cert ends up being named with the namespace-routename.pem unique key so we will overwrite them if they get updated.